jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherBehavioralTest.java

```diff
@@ -53,7 +55,7 @@ void create_withNoServices_shouldNotCreateIngress() {
     // When
     new IngressEnricher(context).create(PlatformMode.kubernetes, klb);
     // Then
-    assertThat(klb.build()).extracting(KubernetesList::getItems).asList().isEmpty();
+    assertThat(klb.build()).extracting(KubernetesList::getItems).asInstanceOf(InstanceOfAssertFactories.LIST).isEmpty();
```

Comment:
```suggestion
    assertThat(klb.build()).extracting(KubernetesList::getItems).asInstanceOf(InstanceOfAssertFactories.list(HasMetadata.class)).isEmpty();
```

Reply:
Pushing pushing a commit with asInstanceOf(InstanceOfAssertFactories.list(HasMetadata.class)) right now for each of these. If you could kindly still respond to my query above I'd appreciate it.

Reply:
In this case it doesn't really matter.
The answer would differ depending on what properties were to be extracted later on from the list (in an hypothetical test).
If you were to access spec (ServiceSpec), then you'd probably want Service. If you were to access metadata (ObjectMeta), then either option would be valid.

Reply:
Thanks for that answer @manusa I figured that, but because from what I've seen before when I was doing a similar issue it used Container.class. Thanks for making this clearer.